File: pythonCode/thresholding.py
import cv2

#Read a gray scale image or read a color image in gray scale format
inputImage = cv2.imread('../assets/putin.jpg', cv2.IMREAD_GRAYSCALE)

#Binary thresholding
th, binaryImage = cv2.threshold(inputImage, 100, 255, cv2.THRESH_BINARY)

#create windows for displaying images
cv2.namedWindow("input image", cv2.WINDOW_NORMAL)
cv2.namedWindow("thresholded image", cv2.WINDOW_NORMAL)

#displaying images
cv2.imshow("input image", inputImage)
cv2.imshow("thresholded image", binaryImage)

#Press esc on keyboard to exit the program
cv2.waitKey(0)

#close all the opened windows
cv2.destroyAllWindows()

File: cPlusPlusCode/thresholding.cpp
#include<opencv2/opencv.hpp>
#include<iostream>

using namespace cv;
using namespace std;

int main(){

    //Read a gray scale image or read a color image in gray scale format
    Mat inputImage = imread("../assets/gray_putin.jpg");

    Mat binaryImage;

    //Set the threshold and maximum values
    double thresh = 100;
    double maxValue = 255;

    //Binary threshold
    threshold(inputImage, binaryImage, thresh, maxValue, THRESH_BINARY);

    //Create windows to display images
    namedWindow("input image", WINDOW_NORMAL);
    namedWindow("binary image", WINDOW_NORMAL);

    //show images
    imshow("input image", inputImage);
    imshow("binary image", binaryImage);

    //Press any key on keyboard to exit the program
    waitKey(0);

    //close the opened windows
    destroyAllWindows();

    return 0;
}